### PR TITLE
Refactoring and fix.

### PR DESCRIPTION
**Pull request by dretax**

### CODE REVIEW

**Comment by dretax** [comment]
@Pecacheu If you don't mind i have done some refactoring.
I am really suggesting your to switch to IntellIJ IDEA instead of Eclipse, and start using maven. (I can help you with that.)

Aside of that I have done fixes, but I haven't been able to create an elevator yet. Seems like the sign is remaining as "Error"

**Comment by dretax** [comment]
Opening only works for metal doors at the moment.
Some API needs to be updated to the current spigot version, but it seems fine now.
[Elevators v2.zip](https://github.com/Pecacheu/Elevators-v2/files/3256897/Elevators.v2.zip)

**Comment by tkf0** [comment]
How difficult would it be to re-enable glass doors? My whole server relies on elevators using doors of said material :D

**Comment by dretax** [comment]
I'll see to It, haven't checked It yet.

**Comment by dretax** [comment]
@tkf0 Update your config.

Following sections are important:
blockList:
  IRON_BLOCK: 8
  GOLD_BLOCK: 10
  EMERALD_BLOCK: 12
  DIAMOND_BLOCK: 15
  LEGACY_STAINED_GLASS: 5
  GLASS: 4

doorBlock: 'GLASS_PANE' # Material used for elevator block-doors.

**Comment by dretax** [comment]
[Elevators v2.zip](https://github.com/Pecacheu/Elevators-v2/files/3269768/Elevators.v2.zip)

**Comment by OverDhill** [comment]
Got the basics working but seen this error in log. 

Error while loading config!

No enum constant org.bukkit.Material.STAINED_GLASS

**Comment by dretax** [comment]
@OverDhill 
image

**Comment by OverDhill** [comment]
Thanks.... it is working great

**Comment by tkf0** [comment]
@dretax I'm so grateful of your efforts, however, I cannot get it to work for my setup.

That was my old config:
```
blockList:
  WOOL: 12
  IRON_BLOCK: 12
  GLASS: 12
doorBlock: 'THIN_GLASS'
```

And here's the one for 1.14:
```
blockList:
  WHITE_WOOL: 12
  IRON_BLOCK: 12
  GLASS: 12

doorBlock: 'GLASS_PANE'
```

As soon as I start my server with the previously working elevator locations, the log says it deleted all of them because they are invalid. Also, I cannot build any new elevators. Attaching the sign with `[Elevator]` does nothing at all.

**Comment by tkf0** [comment]
@dretax I got it working! Thank you!

Would it be possible to fix the doors/glass panes not properly "connecting"/attaching to the adjacent blocks (stone here)?

Here's my current config:

```
blockList:
  WHITE_WOOL: 12
  IRON_BLOCK: 12
  GLASS: 12

doorBlock: 'GLASS_PANE'
```

gp

**Comment by tkf0** [comment]
Also, while just a minor bug, there seems to be an encoding issue:

minorbug

File: Elevators v2/src/com/pecacheu/elevators/Main.java [hunk 1]
```diff
@@ -31,309 +32,548 @@
...
+                            event.setLine(0, Conf.ERROR);
+                            Conf.err("onSignChange:ElevSign:NewElev", "Floor not found!");
+                            return;
+                        }
+                        String eID = Conf.locToString(new Location(fl.world, fl.xMin, 0, fl.zMin));
+                        elev = new Elevator(fl, null, null);
+                        fl.elev = elev;
+                        ind = -1;
+                        Conf.elevators.put(eID, elev);
+                        event.getPlayer().sendMessage("�eElevator created.");
```
Comment (tkf0) [inline Elevators v2/src/com/pecacheu/elevators/Main.java:238]: ...here

**Comment by Smxxthy** [comment]
I noticed that there is a little bug where everytime the elevator starts moving, if you have any doors for the doors on any level, it opens and closes once, maybe twice, really quickly, and the same thing happens when it stops, and it lags my server. Is there a fix for this?

**Comment by Smxxthy** [comment]
I noticed a bug that when the elevator is told to go to another floor, if you are using doors, then the doors open and close really quickly once. This seems to be causing lag for all players.

**Comment by dretax** [comment]
Interesting, I didn't get those encoding issues. I'll take a look at these soon.

**Comment by Pecacheu** [comment]
Ahh I can't believe I didn't accept this pull forever ago, I thought I had. Anyhow someone convinced me to start playing Minecraft again, I'll have a look at it and see how it goes on 1.15.x

**Comment by Pecacheu** [comment]
Well there seem to be a few issues getting it to compile for 1.15, however the pull is closer than it was before.

Unfortunately the formatting of the files is completely messed up, making diff-checking an absolute pain, so I expect it to take a few days to work through everything, however after I get things working I'll accept the pull and release a 1.15 version.

**Comment by dretax** [comment]
Let me know. I dropped MC again. xD

**Pull request merged**